src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/BodyModelBinder.cs

```diff
@@ -65,9 +65,6 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
                 throw new ArgumentNullException(nameof(bindingContext));
             }
 
-            // For compatibility with MVC 5.0 for top level object we want to consider an empty key instead of
-            // the parameter name/a custom name. In all other cases (like when binding body to a property) we
-            // consider the entire ModelName as a prefix.
```

Comment:
But `bindingContext.ModelName` will be the empty string most of the time. You can change the code, but it doesn't solve a problem.

Do you have another suggestion?